### DISCUSSION TITLE
Deploying first contract

### DISCUSSION THREAD

**Discussion by Joaszek** [Cyfrin/foundry-full-course-cu#2097] [Q&A]
I try to deploy SimpleStorage with Injected provider, but when I try to do this I get error:
Gas estimation errored with the following message (see below). The transaction execution will likely fail. Do you want to force sending? 
Returned error: Failed to serialize transaction: toAddressIsNull
[object Object]

![Screenshot from 2024-07-24 15-35-06](https://github.com/user-attachments/assets/8f156401-203b-43c0-885b-72287d86e649)
![Screenshot from 2024-07-24 15-36-05](https://github.com/user-attachments/assets/870792d4-2598-4156-a0c6-ef9785d56fcf)
![Screenshot from 2024-07-24 15-36-25](https://github.com/user-attachments/assets/0bce6271-8da6-45a6-8871-8f55cb82775f)

I see that I have ETH, but when I try to make transaction I get 0. I'm also on the correct chain

**Comment by EngrPips** [comment]
Hello @Joaszek, use `wallet connect` option in remix and retry.

**Comment by Shiva-Sai-ssb** [comment]
Hey @Joaszek,
I see you are facing issues while deploying your smart contract. Are you trying to deploy the contract on Ethereum Sepolia or zkSync Sepolia? If you want to deploy the contract on `Ethereum Sepolia`, make sure to change the network on MetaMask to Ethereum Sepolia.

If you want to deploy the contract on `zkSync Sepolia`, you need to activate the `zkSync plugin` to deploy the contract to zkSync Sepolia. Compile and deploy the contract from the plugin itself. 

Make sure to set the compiler version in the pragma declaration to `0.8.24`, as it only works with that version. You can activate the plugin from the `Plugin Manager`, located in the bottom left corner above the settings on the Remix site.

**Comment by wangshouren7** [comment]
Hi, maybe you need select your test network
![image](https://github.com/user-attachments/assets/16cfe006-3a5b-4d7c-b25d-59d9138922c7)

**Comment by EngrPips** [comment]
Hello @Joaszek, Have you been able to resolve this issue yet?

**Comment by ImFardeenAlam** [comment]
I have the solve for your probelm. You may have connected into the wrong testnet (like ZK Sync Sepolia etc). Rather you have to connect to "Sepolia" testnet.

**Reply by EngrPips** [reply]
Interesting.
